aiida_quantumespresso/parsers/pp.py

```diff
@@ -71,34 +70,44 @@ def parse(self, **kwargs):
         except (IOError, OSError):
             return self.exit_codes.ERROR_OUTPUT_STDOUT_READ
 
-        # The post-processed data should have been written to file, either in the retrieved or temp list
-        filename_data = PpCalculation._FILEOUT
-        if filename_data in self.retrieved.list_object_names():  # Retrieved list case
-            try:
-                data_raw = self.retrieved.get_object_content(filename_data)
-            except (IOError, OSError):
-                return self.exit_codes.ERROR_DATAFILE_READ
-        elif temp_folder_path is not None:  # Temp list case
-            data_file_path = os.path.join(temp_folder_path, filename_data)
-            if os.path.isfile(data_file_path):
-                try:
-                    with open(data_file_path, 'r') as fhandle:
-                        data_raw = fhandle.read()
-                except (IOError, OSError):
-                    return self.exit_codes.ERROR_DATAFILE_READ
-            else:
-                return self.exit_codes.ERROR_OUTPUT_DATAFILE_MISSING
+        data_raw = []
+
+        # Currently all plot output files should start with the `filplot` as prefix. If only one file was produced the
+        # prefix is the entire filename, but in the case of multiple files, there will
+        filename_prefix = PpCalculation._FILPLOT
+        filename_suffix = PpCalculation._FILEOUT
+
+        # How to get the output filenames and how to open them, depends on whether they will have been retrieved in the
+        # `retrieved` output node, or in the `retrieved_temporary_folder`. Instead of having a conditional with almost
+        # the same loop logic in each branch, we apply a somewhat dirty trick to define an `opener` which is a callable
+        # that will open a handle to the output file given a certain filename.
+        if retrieve_temporary_list:
```

Comment:
Might be worth mentioning here that there can either be _all_ temporary files or _all_ retrieved, not a mix of the two.

I think this is a decent choice, but had to go check in the calculation that this is what's implemented there, too.